pkg/graveler/graveler.go

```diff
@@ -1678,6 +1688,11 @@ func (g *Graveler) Delete(ctx context.Context, repository *RepositoryRecord, bra
 		return ErrWriteToProtectedBranch
 	}
 
+	isProtected = g.readOnlyRepositoryManager.IsBlocked(repository)
+	if isProtected {
+		return ErrWriteToReadOnlyRepository
+	}
```

Comment:
Also, the use of isProtected is confusing, protected it's for protected branches, this is a read-only repository